database/api_test.go

```diff
@@ -121,6 +124,185 @@ func (matcher *queryMatcher) String() string {
 	return "*qb.SelectQuery"
 }
 
+func Test_SelectWithTotal(t *testing.T) {
+	assert := assert1.New(t)
+	ctrl := gomock.NewController(t)
+	api := NewMockAPI(ctrl)
+
+	var (
+		err    error
+		total  int
+		target TestRecordCollection
+		limit = 0
+		offset = 0
+	)
+
+	query := qb.Select(MetaTestRecord.ID).From(MetaTestRecord)
+
+	//limit equal zero 
+	api.EXPECT().Count(MetaTestRecord, query).Return(int32(2), nil)
+	
+	_, total, err = SelectWithTotal(api, MetaTestRecord, target, query, limit, offset)
+
+	assert.NoError(err)
+	assert.Equal(2, total)
```

Comment:
Create these with their own scopes so that failures are easier to find. Also, make sure the assert and require you are using have the correct t for the test that is running or it will be unclear what test is failing. You should get lint errors that require should not be used to check errors. Make sure you have the following or better:

```
$ go version
go version go1.21.6 linux/amd64
$ golangci-lint version
golangci-lint has version v1.55.2 built with go1.21.6 from (unknown, mod sum: "h1:yllEIsSJ7MtlDBwDJ9IMBkyEUz2fYE0b5B8IUgO1oP8=") on (unknown)
```

```suggestion
	t.Run("limit equal zero", func(t *testing.T) {
		assert := assert1.New(t)
		require := _require.New(t)
		ctrl := gomock.NewController(t)
		api := NewMockAPI(ctrl)
		
		api.EXPECT().Count(MetaTestRecord, query).Return(int32(2), nil)
	
		_, total, err = SelectWithTotal(api, MetaTestRecord, target, query, limit, offset)

		require.NoError(err)
		assert.Equal(2, total)
	}
```

Reply:
Sorry my comment was unclear and had a typo. When testing error conditions always use 'require'. If you are modifying the file, you should make sure all the lint in that file is taken care of, even if you would not otherwise have touched that scope.